--- ricty_generator.sh.orig	2011-05-18 02:02:00.000000000 +0900
+++ ricty_generator.sh	2011-05-18 16:33:27.000000000 +0900
@@ -269,7 +269,7 @@
     OverlapIntersect()
     Copy(); Select(0u2015); Paste()
     # post proccess
-    SelectWorthOutputting(); RemoveOverlap(); RoundToInt()
+    SelectWorthOutputting(); RoundToInt()
     # generate
     Generate(fontfamily+"-"+fontstyle_list[i]+".ttf", "", 0x84)
 i += 1; endloop
